acceptancetests/assess_persistent_storage.py

```
@@ -27,6 +33,116 @@
+    """This function tests charm deployment and removal with single
+       persistent storage unit.
+
+       Steps taken to the test:
+       - Deploy dummy-storage charm with a single block storage unit (ebs)
+         and a single filesystem storage unit (rootfs).
+       - Check charm status once the deployment is done.
+       - Check charm storage list once the deployment is done.
+       - Remove the deployed charm, the output should indicate that the
+         rootfs storage unit will be removed, ebs storage unit will be
+         detached.
+       - Check storage list again to ensure ebs storage units really exists.
+
+       :param client: ModelClient object to deploy the charm on.
+    """
+
+    charm_name = 'dummy-storage'
+    charm_path = local_charm_path(
+        charm=charm_name, juju_ver=client.version)
+    log.info(
+        '{} is going to be deployed with 1 ebs block storage unit and '\
+        '1 rootfs filesystem storage unit.'.format(charm_name))
+    log.info(
+        'juju deploy ${JUJU_REPOSITORY}/charms/dummy-storage '\
+        '--storage single-blk=ebs --storage single-fs=rootfs')
+    # client.deploy() doesn't support repeat parameters,
+    # although juju deploy supports it
+    #client.deploy(
+    #    charm_path,
+    #    storage='single-blk=ebs',
+    #    storage='single-fs=rootfs')
+    client.get_juju_output(
+        'deploy', '/home/noname/work/Canonical/workbench/repository/charms/dummy-storage',
+        '--storage', 'single-blk=ebs',
+        '--storage', 'single-fs=rootfs',
+        include_e=False)
+    client.wait_for_started()
+    client.wait_for_workloads()
+
+    log.info('juju status --format json')
+    status_output = json.loads(
+        client.get_juju_output('status', '--format', 'json', include_e=False))
+    app_status = status_output['applications'][charm_name]['application-status']['current']
+
+    if app_status != 'active':
+        log.error(
```

I'd like to see these functioned out for readability. In addition, you should raise an exception, not log an error and then sys.exit().
```
if app_status != 'active':
            raise JujuAssertionError('App Status is incorrect. Found: {}\nExpected: {}'.format(found, expected))
```

In addition, I modified the error. "charm is not functioning properly is not a good error message, nor does it describe your assertion". Finally, is this even required? Given your wait_for_*, you shouldn't arrive at this point without a valid status right?